### ISSUE TITLE
Crash in test_slice on Intel CL

### ISSUE THREAD

**Issue by alexfikl** [inducer/pyopencl#640]
**Describe the bug**

`test_slice` still seems to be crashing on Intel CL even after #638.

**To Reproduce**

https://github.com/inducer/pyopencl/actions/runs/3273678656/jobs/5386201682

**Environment (please complete the following information):**
 - OS: Windows, Linux
 - ICD Loader and version: ocl-icd 2.3.1
 - ICD and version: intel-opencl-rt 2022.1.0
 - CPU/GPU: Intel(R) Xeon(R) Platinum 8370C CPU
 - Python version: 3.10.8
 - PyOpenCL version: master

cc: @isuruf

**Comment by inducer** [comment]
Cannot reproduce locally with the following conda env: https://gist.github.com/inducer/6009379264cd7c23ee0bdb9306a18a3b

**Comment by isuruf** [comment]
Fixed in https://github.com/inducer/pyopencl/pull/641

@inducer, we need to run the test a few times to reproduce the failure.

**Comment by alexfikl** [comment]
I managed to hit this again in
https://github.com/inducer/pyopencl/actions/runs/3388095221/jobs/5629660314
Any sense if it's related or a new issue?

**Comment by inducer** [comment]
IDK. The Intel CL runtime is unreasonably buggy. I don't know how much more time I'd like to waste chasing their regressions. I've banned another version in #653, maybe that'll help?

**Comment by inducer** [comment]
FWIW, the `test_slice` thing was never a pyopencl issue AFAICT.

**Comment by alexfikl** [comment]
> IDK. The Intel CL runtime is unreasonably buggy. I don't know how much more time I'd like to waste chasing their regressions. I've banned another version in #653, maybe that'll help?

Is this their CPU runtime? Either way, agreed it's not worth that much effort at this point.

**Comment by inducer** [comment]
Yeah. Their GPU runtime is of far higher quality in my experience.
